ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/PartitionEvaluator.java

```diff
@@ -49,6 +49,9 @@ interface Weight extends Comparable<Weight> {}
    */
   boolean addFile(DataFile dataFile, List<ContentFile<?>> deletes);
 
+  /** After table file scan, confirm whether the segment files really needs to be rewritten. */
+  void globalEvaluate();
```

Comment:
Done.
Removed `globalEvaluate` in this interface and moved to `isNecessary()`.
As another comment described, I added `undersizedSegmentFiles.clear();` , which not need add some flag to prevent adding more files after invoking it. 
WDYT?